python/desc/imsim/camera_readout.py

```diff
@@ -273,12 +291,19 @@ def get_amplifier_hdu(self, amp_name):
             Image HDU with the pixel data and header keywords
             appropriate for the requested sensor segment.
         """
-        hdu = fits.ImageHDU(data=self.amp_images[amp_name].getArray(),
-                            do_not_scale_image_data=True)
+        hdu = fits.ImageHDU(data=self.amp_images[amp_name].getArray().astype(np.int32))
 
         # Copy keywords from eimage primary header.
-        for key in self.eimage[0].header.keys():
-            hdu.header[key] = self.eimage[0].header[key]
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            for key in self.eimage[0].header.keys():
+                try:
+                    hdu.header[key] = self.eimage[0].header[key]
+                except ValueError as eobj:
```

Comment:
Does this mean in those cases the copy will fail and we won't have the header? Should we "fix" them instead or is this a rare occurrence?

Reply:
This only arises for *phosim* eimage files.  We're not going to fixing phosim's file output, but I think we will want to be able to run this script on those eimage files in order to get "amplifier" files that look like real camera data.